_posts/2019-01-31-CocoaPods-1.7.0-beta.markdown

```diff
@@ -0,0 +1,205 @@
+
+Finally, a warning will be displayed during `lint` that encourages pod authors to migrate away from using the `.swift-version` file and in a future major release of CocoaPods we plan to completely remove support for it.
+
+### App Specs
+
+With the recently introduced [test specs](http://blog.cocoapods.org/CocoaPods-1.3.0/) we were able to build a platform for us to expand on and introduce different types of specifications that can be provided by a pod author. With 1.7.0 we are introducing _app specs_ which allow pod authors to describe an application from within their `podspec`.
+
+App specs can help pod authors in various ways, for example, they can be used to ship a sample app along side their pod as a tutorial on how consumers can integrate it to their own respective app.
+
+App specs progress forward the concept of 'Isolated Development' in which a `podspec` can be used as the single piece of information needed to generated an entire (throwable) workspace to develop on.
+
+Here's an example of an app spec declaration:
+
+```ruby
+Pod::Spec.new do |s|
+  s.name         = 'CoconutLib'
+  s.version      = '1.0'
+  s.authors      = 'Coconut Corp', { 'Monkey Boy' => 'monkey@coconut-corp.local' }
+  s.homepage     = 'http://coconut-corp.local/coconut-lib.html'
+  s.summary      = 'Coconuts For the Win.'
+  s.description  = 'All the Coconuts'
+  s.source       = { :git => 'http://coconut-corp.local/coconut-lib.git', :tag => 'v1.0' }
+  s.license      = {
+    :type => 'MIT',
+    :file => 'LICENSE',
+    :text => 'Permission is hereby granted ...'
+  }
+
+  s.source_files = 'Sources/*.{h,m}'
+
+  s.app_spec 'SampleApp' do |app_spec|
+    app_spec.source_files = 'Sample/*.swift'
+  end  
+end
+```
+
+App specs can leverage the majority of the CocoaPods spec DSL to declare dependencies, build configuration settings, etc. 
+
+We hope in the future to promote app specs to a top-level concept in which consumer applications are described as an app spec completely eliminating the need of an `.xcodeproj` in order to integrate with CocoaPods.
+
+### Multiple Xcodeproj Generation
+
+Historically, CocoaPods has always generated a single `Pods.xcodeproj` that contains all of the required targets and build settings for your project to compile. Using just one project that incorporates your entire `Podfile` works just fine for smaller projects; however, as your project grows, so will the size of your `Pods.xcodeproj` file.
+
+The larger the `Pods.xcodeproj` file, the longer Xcode has to spend parsing its contents, which we have discovered leads to a degraded Xcode experience. Instead of putting all of your targets into one monolithic Xcode project, we’ve noticed some considerable performance improvements for larger CocoaPods projects by integrating each pod as its own separate Xcode project.
```

Comment:
This makes it sounds like `Pods.xcodeproj` is going away - maybe add some implementation details explaining that each dependency will have a project that get's added as a subproject to `Pods.xcodeproj`. Not sure how to word that in a better way though